crates/cli/src/package.rs

```diff
@@ -35,17 +38,41 @@ pub async fn fetch_package_version_directly<P: Into<PathBuf>>(
 ) -> Result<PackageVersion, PackageManagerError> {
     let package_version =
         PackageVersion::fetch_from_registry(name, version, http_client, &config.registry).await?;
-    internal_fetch(&package_version, config, symlink_path).await?;
+    internal_fetch(package_cache, &package_version, config, symlink_path).await?;
     Ok(package_version.to_owned())
 }
 
 async fn internal_fetch<P: Into<PathBuf>>(
+    package_cache: &PackageCache,
     package_version: &PackageVersion,
     config: &Npmrc,
     symlink_path: P,
 ) -> Result<(), PackageManagerError> {
     let store_folder_name = package_version.to_store_name();
 
+    let saved_path = config
+        .virtual_store_dir
+        .join(store_folder_name)
+        .join("node_modules")
+        .join(&package_version.name);
+
+    if let Some(mut receiver) = package_cache.get_mut(&saved_path) {
+        // TODO: is this loop necessary?
```

Comment:
If we use a `Mutex<Receiver<PackageState>>` we don't have to have a loop here. Similar to: https://medium.com/@polyglot_factotum/rust-concurrency-patterns-condvars-and-locks-e278f18db74f

Reply:
This is not what my question was referring to. I am wonder whether I should remove the whole code block inside loop (leaving only `return Ok(())`) or keep it. This would only be clear after further work. Hence the TODO.

Reply:
If the package is `InProgress` we need to wait, since after this function, we call package import functions to link the store files with node_modules.